src/Native/Runtime/thread.cpp

```diff
@@ -371,6 +372,14 @@ bool Thread::IsCurrentThread()
 
 void Thread::Destroy()
 {
+#ifndef PLATFORM_UNIX
```

Comment:
Why can't we do this in C#? It looks pretty unfortunate to have counter of thread pool threads in redhawk.

Reply:
I considered a finalizer but thought that it could be significantly delayed. Perhaps it won't be a big issue for tracking the thread count, it may not be accurate but it would still be somewhat correct.

It would also be possible to move most of the code if on the destroy path it would be possible to run some managed code. Maybe a possibility is to signal the finalizer thread to run some code and adjust the counts, I'll see if that would be possible.

Reply:
Hmm not sure the finalizer thread can be notified on thread detach, as it would need to anyway track whether it's a thread pool thread and track some counter to adjust the actual count, which probably wouldn't be much different from this.

I could use a thread-static variable with a finalizer, with the caveat that it may not detect threads that exit quickly enough.

Reply:
This API is approximation anyway when we do not own the threadpool. It is really tracking "How many threads in the OS threadpool happened to run managed code and have not exited yet". I think it is ok for it to the tracking to be delayed.